deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/AbstractLSTM.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    @Override
+    public double getL1ByParam(String paramName) {
+        switch (paramName) {
+            case LSTMParamInitializer.INPUT_WEIGHT_KEY:
+            case LSTMParamInitializer.RECURRENT_WEIGHT_KEY:
+                return l1;
+            case LSTMParamInitializer.BIAS_KEY:
+                return l1Bias;
+            default:
+                throw new IllegalArgumentException("Unknown parameter name: \"" + paramName + "\"");
+        }
+    }
+
+    @Override
+    public double getL2ByParam(String paramName) {
+        switch (paramName) {
+            case LSTMParamInitializer.INPUT_WEIGHT_KEY:
+            case LSTMParamInitializer.RECURRENT_WEIGHT_KEY:
+                return l2;
+            case LSTMParamInitializer.BIAS_KEY:
+                return l2Bias;
+            default:
+                throw new IllegalArgumentException("Unknown parameter name: \"" + paramName + "\"");
+        }
+    }
+
+    @Override
+    public double getLearningRateByParam(String paramName) {
+        switch (paramName) {
+            case LSTMParamInitializer.INPUT_WEIGHT_KEY:
+            case LSTMParamInitializer.RECURRENT_WEIGHT_KEY:
+                return learningRate;
+            case LSTMParamInitializer.BIAS_KEY:
+                if (!Double.isNaN(biasLearningRate)) {
+                    //Bias learning rate has been explicitly set
+                    return biasLearningRate;
+                } else {
+                    return learningRate;
+                }
+            default:
+                throw new IllegalArgumentException("Unknown parameter name: \"" + paramName + "\"");
+        }
+    }
+
+    @AllArgsConstructor
+    @NoArgsConstructor
+    public static abstract class Builder<T extends Builder<T>> extends BaseRecurrentLayer.Builder<T> {
+
+        protected double forgetGateBiasInit = 1.0;
+        protected IActivation gateActivationFn = new ActivationSigmoid();
+
+        /** Set forget gate bias initalizations. Values in range 1-5 can potentially
+         * help with learning or longer-term dependencies.
+         */
+        public T forgetGateBiasInit(double biasInit) {
+            this.forgetGateBiasInit = biasInit;
+            return (T)this;
+        }
+
+        /**
+         * Activation function for the LSTM gates.
+         * Note: This should be bounded to range 0-1: sigmoid or hard sigmoid, for example
+         *
+         * @param gateActivationFn Activation function for the LSTM gates
+         */
+        public T gateActivationFunction(String gateActivationFn) {
+            return (T)gateActivationFunction(Activation.fromString(gateActivationFn));
+        }
+
+        /**
+         * Activation function for the LSTM gates.
+         * Note: This should be bounded to range 0-1: sigmoid or hard sigmoid, for example
+         *
+         * @param gateActivationFn Activation function for the LSTM gates
+         */
+        public T gateActivationFunction(Activation gateActivationFn) {
```

Comment:
Maybe add a validation warning here? It shouldn't be hard to detect which activation functions have this property.